Client/Frontend/Browser/Favorites/FavoritesViewController.swift

```diff
@@ -279,7 +279,9 @@ extension FavoritesViewController: UICollectionViewDataSource, UICollectionViewD
             var urlChildren: [UIAction] = [openInNewTab]
             if !PrivateBrowsingManager.shared.isPrivateBrowsing {
                 let openInNewPrivateTab = UIAction(title: Strings.openNewPrivateTabButtonTitle, image: nil, identifier: nil, discoverabilityTitle: nil) { _ in
-                    self.action(bookmark, .opened(inNewTab: true, switchingToPrivateMode: true))
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
+                        self.action(bookmark, .opened(inNewTab: true, switchingToPrivateMode: true))
```

Comment:
Why it crashes without it?

Reply:
The action occurs which removes the NTP/Overlay, when that happens and the context menu tries to dismiss its preview, the view doesn't exist to animate back to, which crashes here: https://github.com/brave/brave-ios/blob/184fc8f4864636e7dbbd03a5c45c3524d9b7273a/Client/Frontend/Browser/Favorites/FavoritesViewController.swift#L310

Reply:
Sounds good, can add a comment about it